executor/src/witgen/machines/double_sorted_witness_machine.rs

```diff
@@ -183,35 +210,47 @@ impl<'a, T: FieldElement> Machine<'a, T> for DoubleSortedWitnesses<T> {
             step.push(s);
             value.push(o.value);
 
-            is_write.push(o.is_write.into());
-            is_read.push((!o.is_write).into());
+            is_normal_write.push(o.is_normal_write.into());
+            is_bootloader_write.push(o.is_bootloader_write.into());
+            is_read.push((!o.is_write()).into());
         }
         if addr.is_empty() {
             // No memory access at all - fill a first row with something.
-            addr.push(0.into());
+            addr.push(-T::one());
```

Comment:
This is necessary if the memory machine was compiled with the bootloader, because then the `LAST * (1 - m_change) * (m_addr + 1) = 0` constraint allows us to set `m_change` to 0 everywhere if we have zero memory accesses.

In the "normal" memory machine, this does not harm.